### PR TITLE
Add "Form Inside List" sample for email

### PR DESCRIPTION
**Pull request by fstanis**
An email sample to demonstrate how to load the initial data using `amp-list` and then use `amp-form` to update part of it based on user input.

### CODE REVIEW

File: src/amphtml-email/20_Samples/Form_Inside_List.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+based on a user's subsequent input in an [`amp-form`](https://www.ampproject.org/docs/reference/components/amp-form).
+-->
+
+<!-- -->
+<!doctype html>
+<html ⚡4email>
+<head>
+  <meta charset="utf-8">
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <script async custom-element="amp-list" src="https://cdn.ampproject.org/v0/amp-list-0.1.js"></script>
```
Comment (kul3r4) [inline src/amphtml-email/20_Samples/Form_Inside_List.html]: can you comment for each library what is used for? Similar to [this](https://github.com/ampproject/amp-by-example/blob/master/src/30_Advanced/Autosuggest_form.html)

File: src/amphtml-email/20_Samples/Form_Inside_List.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    {{#nickname}}
+    <p>The nickname you chose is {{nickname}}.</p>
+    {{/nickname}}
+    {{^nickname}}
+    <p>No nickname selected.</p>
+    {{/nickname}}
+  </template>
+
+  <!--
+    Then define the template that the list will use, which will contain an `amp-form`
```
Comment (kul3r4) [inline src/amphtml-email/20_Samples/Form_Inside_List.html]: replace "Then define the template that the list will use" with "Then define the template that `amp-list` will use"

File: src/amphtml-email/20_Samples/Form_Inside_List.html [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    and reference the template we created above.
+  -->
+  <template id="list-template" type="amp-mustache">
+    <p>Hello world!</p>
```
Comment (sebastianbenz) [inline src/amphtml-email/20_Samples/Form_Inside_List.html]: remove this line

File: src/amphtml-email/20_Samples/Form_Inside_List.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    ~-->
+    <div id="form-placeholder">
+      <p>No nickname selected.</p>
+    </div>
+  </template>
+
+  <!--
+    Finally, use the `template` attribute in `amp-list` to refer to the template above.
+  -->
+  <amp-list template="list-template" items="." single-item src="https://ampbyexample.com/echo" layout="fixed-height" height="100"></amp-list>
```
Comment (sebastianbenz) [inline src/amphtml-email/20_Samples/Form_Inside_List.html]: Can you explain that the `src` is just a dummy endpoint needed to render the initial template. 

Sidenote: I find it a bit unintuitive to use an amp-list here to render the form (you could simply server-side render the form). Maybe explain this.

**Comment by sebastianbenz** [comment]
@fstanis ping

**Pull request closed without merging**